Analysis/Tasks/PWGDQ/dileptonEE.cxx

```diff
@@ -71,185 +89,268 @@ void DefineHistograms(o2::framework::OutputObj<HistogramManager> histMan, TStrin
   {
+    DefineCuts();
+
     fValues = new float[VarManager::kNVars];
     VarManager::SetDefaultVarNames();
-    fHistMan.setObject(new HistogramManager("analysisHistos", "aa", VarManager::kNVars));
+    fHistMan = new HistogramManager("analysisHistos", "analysisHistos", VarManager::kNVars);
     fHistMan->SetUseDefaultVariableNames(kTRUE);
     fHistMan->SetDefaultVarNames(VarManager::fgVariableNames, VarManager::fgVariableUnits);
 
-    DefineHistograms(fHistMan, "TrackBarrel_BeforeCuts;TrackBarrel_AfterCuts;"); // define all histograms
-    VarManager::SetUseVars(fHistMan->GetUsedVars());                             // provide the list of required variables so that VarManager knows what to fill
+    TString cutNames = "TrackBarrel_BeforeCuts;";
+    for (int i = 0; i < gNTrackCuts; i++) {
+      cutNames += Form("TrackBarrel_%s;", fTrackCuts[i].GetName());
+    }
 
-    DefineCuts();
+    DefineHistograms(fHistMan, cutNames.Data());     // define all histograms
+    VarManager::SetUseVars(fHistMan->GetUsedVars()); // provide the list of required variables so that VarManager knows what to fill
+    fOutputList.setObject(fHistMan->GetMainHistogramList());
   }
 
   void DefineCuts()
   {
-    fTrackCut = new AnalysisCompositeCut(true); // true: use AND
-    AnalysisCut* cut1 = new AnalysisCut();
-    cut1->AddCut(VarManager::kPt, 0.2, 10.0);
-    cut1->AddCut(VarManager::kEta, -0.8, 0.8);
-    cut1->AddCut(VarManager::kTPCchi2, 0.0, 4.0);
-    cut1->AddCut(VarManager::kITSchi2, 0.0, 5.0);
-    cut1->AddCut(VarManager::kITSncls, 3.5, 7.5);
-    cut1->AddCut(VarManager::kTPCncls, 69.5, 159.5);
-    cut1->AddCut(VarManager::kTrackDCAxy, -1.0, +1.0);
-    cut1->AddCut(VarManager::kTrackDCAz, -3.0, +3.0);
-
-    cut1->AddCut(VarManager::kTPCsignal, 70, 90, false);                               //exclude = false
-    cut1->AddCut(VarManager::kTPCsignal, 75, 90, false, VarManager::kPin, 2.0, 1e+10); //exclude = false
-    //cut1->AddCut(VarManager::kTOFnSigmaEl, -3, +3, false); //exclude = false
-    //fTrackCut->AddCut(cut1);
-
-    AnalysisCompositeCut* pidcut = new AnalysisCompositeCut(false); // false : use OR
-    AnalysisCut* pidcut_high = new AnalysisCut("pidcut_high", "e high");
-    pidcut_high->AddCut(VarManager::kTPCsignal, 75, 90, false, VarManager::kPin, 2.0, 1e+10, false); //exclude = false
-    pidcut_high->AddCut(VarManager::kTPCsignal, 70, 90, false);                                      //exclude = false
-
-    AnalysisCut* pidcut_rejPr = new AnalysisCut("pidcut_rejPr", "proton rejection");
-    TF1* f1minPr = new TF1("f1minPr", "[0]+[1]*x", 0, 10);
-    f1minPr->SetParameters(170, -100);
-    TF1* f1maxPr = new TF1("f1maxPr", "[0]+[1]*x", 0, 10);
-    f1maxPr->SetParameters(175, -75);
-    pidcut_rejPr->AddCut(VarManager::kTPCsignal, f1minPr, f1maxPr, true, VarManager::kPin, 0.8, 1.4, false); //exclude = false
+    AnalysisCut* commonCuts = new AnalysisCut();
+    commonCuts->AddCut(VarManager::kPt, 0.2, 10.0);
```

Comment:
The quality cuts below are rather standard quality cuts for DQ analysis. You can use the existing predefined cuts in the library. If some are missing there, they can be added.